pkg/cloudprovider/servicelb.go

```diff
@@ -219,16 +239,37 @@ func (k *k3s) getDaemonSet(svc *core.Service) (*apps.DaemonSet, error) {
 // getStatus returns a LoadBalancerStatus listing ingress IPs for all ready pods
 // matching the selected service.
 func (k *k3s) getStatus(svc *core.Service) (*core.LoadBalancerStatus, error) {
-	pods, err := k.podCache.List(k.LBNamespace, labels.SelectorFromSet(map[string]string{
+	var readyNodes map[string]bool
+
+	if servicehelper.RequestsOnlyLocalTraffic(svc) {
+		readyNodes = map[string]bool{}
```

Comment:
Is this necessary if you `make` the map on line 242?

Reply:
I want to intentionally leave it nil if `RequestsOnlyLocalTraffic` is false; since the different between a nil map and a map with no nodes does matter down at line 324